helix-term/src/commands.rs

```diff
@@ -659,36 +668,79 @@ fn goto_line_start(cx: &mut Context) {
     )
 }
 
+fn goto_first_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 0);
+}
+
+fn goto_second_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 1);
+}
+
+fn goto_third_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 2);
+}
+
+fn goto_fourth_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 3);
+}
+
+fn goto_fifth_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 4);
+}
+
+fn goto_sixth_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 5);
+}
+
+fn goto_seventh_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 6);
+}
+
+fn goto_eight_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 7);
+}
+
+fn goto_ninth_buffer(cx: &mut Context) {
+    goto_buffer_by_idx_impl(cx.editor, 8);
+}
+
 fn goto_next_buffer(cx: &mut Context) {
-    goto_buffer(cx.editor, Direction::Forward);
+    goto_buffer_by_direction(cx.editor, Direction::Forward)
 }
 
 fn goto_previous_buffer(cx: &mut Context) {
-    goto_buffer(cx.editor, Direction::Backward);
+    goto_buffer_by_direction(cx.editor, Direction::Backward)
+}
+
+fn goto_buffer_by_direction(editor: &mut Editor, dir: Direction) {
+    let curr_doc = &view!(editor).doc;
+    let buffs_len = editor.documents.len();
+    let curr_i = editor
+        .documents
+        .keys()
+        .position(|d| d == curr_doc)
+        .expect("current document was not in documents");
+
+    let new_i = match dir {
+        Direction::Forward if curr_i < buffs_len - 1 => curr_i + 1,
+        Direction::Forward => 0, // Would be out of bounds, wrap to front.
+        Direction::Backward if curr_i > 0 => curr_i - 1,
+        Direction::Backward => buffs_len - 1, // Would be out of bounds, wrap to back.
+    };
+
+    goto_buffer_by_idx_impl(editor, new_i);
 }
 
-fn goto_buffer(editor: &mut Editor, direction: Direction) {
-    let current = view!(editor).doc;
```

Comment:
I don't think it's necessary to change the body of the old `goto_buffer` function - it has the same behavior before and after except that going to the previous buffer skips from the opposite direction

Reply:
I mainly replaced the body of the function because I saw nearly identical code in both branches of the match statement. That just triggered my refactoring sense, lol.
After looking into it, using an iterator the way it is used here may be hard to understand / overkill for such a simple task.

> except that going to the previous buffer skips from the opposite direction

I'm honestly not certain what you mean by this. `buffer-next` goes "to the right", and `buffer-previous` goes "to the left". Have I missed something that the original implementation does?